File: solutions/binary-search/binary-search.go
// @Title: Binary Search
// @Author: colinjxc
// @Date: 2022-02-02T20:41:45+08:00
// @URL: https://leetcode-cn.com/problems/binary-search


func search(nums []int, target int) int {
    j, k := 0, len(nums)
    for j < k {
        m := (j + k) >> 1
        if nums[m] < target {
            j = m + 1
        } else if (nums[m] > target) {
            k = m
        } else {
            return m
        }
    }
    return -1
}

File: solutions/binary-search/binary-search.py
// @Title: Binary Search
// @Author: colinjxc
// @Date: 2022-01-26T01:16:39+08:00
// @URL: https://leetcode-cn.com/problems/binary-search


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums)
        while i < j:
            m = (i + j)//2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                i = m + 1
            else:
                j = m

        return -1

